test/lib/expr_test_utils.cc

```diff
@@ -389,6 +389,121 @@ std::pair<evaluation_inputs, std::unique_ptr<evaluation_inputs_data>> make_evalu
     return std::pair(std::move(inputs), std::move(data));
 }
 
+mock_database_impl::mock_database_impl(schema_ptr table_schema) {
+    _table_schema = table_schema;
+    _keyspace_metadata = data_dictionary::keyspace_metadata::new_keyspace(
+        _table_schema->ks_name(), "MockReplicationStrategy", {}, false, {_table_schema});
+}
+
+std::pair<data_dictionary::database, std::unique_ptr<mock_database_impl>> mock_database_impl::make(
+    schema_ptr table_schema) {
+    std::unique_ptr<mock_database_impl> mock_db = std::make_unique<mock_database_impl>(table_schema);
+    data_dictionary::database db = make_database(mock_db.get(), nullptr);
+    return std::pair(std::move(db), std::move(mock_db));
+}
+
+std::optional<data_dictionary::keyspace> mock_database_impl::try_find_keyspace(data_dictionary::database db,
+                                                                               std::string_view name) const {
+    if (_table_schema->ks_name() == name) {
+        return make_keyspace(this, nullptr);
+    }
+    return std::nullopt;
+}
+
+std::vector<data_dictionary::keyspace> mock_database_impl::get_keyspaces(data_dictionary::database db) const {
+    return {make_keyspace(this, nullptr)};
+}
+
+std::vector<data_dictionary::table> mock_database_impl::get_tables(data_dictionary::database db) const {
+    return {make_table(this, nullptr)};
+}
+
+std::optional<data_dictionary::table> mock_database_impl::try_find_table(data_dictionary::database db,
+                                                                         std::string_view ks,
+                                                                         std::string_view tab) const {
+    if (_table_schema->ks_name() == ks && _table_schema->cf_name() == tab) {
+        return make_table(this, nullptr);
+    }
+    return std::nullopt;
+}
+
+std::optional<data_dictionary::table> mock_database_impl::try_find_table(data_dictionary::database db,
+                                                                         table_id id) const {
+    if (_table_schema->id() == id) {
+        return make_table(this, nullptr);
+    }
+    return std::nullopt;
+}
+
+const secondary_index::secondary_index_manager& mock_database_impl::get_index_manager(data_dictionary::table t) const {
+    throw std::bad_function_call();
```

Comment:
Note: we will have to move the index stuff into data_dictionary later. Out of scope for now.